test/test_optim.py

```diff
@@ -566,27 +592,14 @@ def test_adam(self):
                  lambda opt: ReduceLROnPlateau(opt)],
                 constructor_accepts_maximize=True
             )
+            self._test_complex_2d(optimizer)
```

Comment:
I'm a bit surprised we're rolling out a custom test here. Why can't `_test_complex_optimizer()` be re-used?